src/tools.cpp

```diff
@@ -9,6 +9,7 @@
 #include <iomanip>
 #include <sstream>
 #include <stdexcept>
+#include <cmath>
```

Comment:
Can you move this include to the top? I try to keep the blocks of includes sorted alphabetically.